### PR TITLE
JS docs

### PR DESCRIPTION
**Pull request by NeerjaGoyalNHS**
## Description
JS docs for CaasFeedAddRecordsLambda

## Type of changes

- [x] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 1]
```diff
@@ -108,7 +117,17 @@ export const handler = async (event) => {
   return "Exiting Lambda";
 };
 
-// METHODS
+/**
+ * Reads a CSV file from S3.
+ * @async
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:122]: Small one, can you keep a space between the function definition and other declarations

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 2]
```diff
@@ -248,7 +321,14 @@ export const checkDynamoTable = async (
   }
 };
 
-// returns formatted dynamodb json to be uploaded or rejected object
+/**
+ * @async
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:325]: This one is missing a definition

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 3]
```diff
@@ -280,8 +360,16 @@ const generateRecord = async (record, client) => {
   return await formatDynamoDbRecord(record);
 };
 
-// returns unique participant id using NHS regex
-// unique in both population table and this stream
+/**
+ * Generates unique participant id using NHS regex
+ * unique in both population table and this stream
+ * @async
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:366]: space here as well please to separate definition and other declarations

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 4]
```diff
@@ -399,7 +506,18 @@ export const getItemFromTable = async (dbClient, table, ...keys) => {
   return response;
 };
 
-// returns successful response if attribute doesn't exist in table
+/**
+ * This function allows the user to query against DynamoDB.
+ * @async
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:511]: Here too

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 5]
```diff
@@ -436,7 +554,19 @@ export const lookUp = async (dbClient, ...params) => {
   return SUCCESSFULL_REPSONSE;
 };
 
-// returns response from batch write to dynamodb table
+//
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:557]: trailing comment by accident?

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 6]
```diff
@@ -26,7 +27,15 @@ const UNSUCCESSFULL_REPSONSE = 400;
 
 let participantIdStore = {};
 
-// Lambda Entry Point
+/**
+ * AWS Lambda handler to load participant info for participants with result - CSD.
```
Comment (chenglawson1) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:31]: Load elligible population, not participants with CSD result.

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 7]
```diff
@@ -26,7 +27,15 @@ const UNSUCCESSFULL_REPSONSE = 400;
...
+ *
+ * @async
+ * @function handler
+ * @param {Object} event - The event triggering the Lambda.
+ * @returns {Promise<Object>} - A promise that resolves to the response object containing participant info or an error message.
```
Comment (chenglawson1) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:36]: I think the promise resolves to a string "Exiting Lambda".

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 8]
```diff
@@ -145,6 +177,17 @@ export const pushCsvToS3 = async (bucketName, key, body, client) => {
...
 
+/**
+ * Processes Csv data read in from the S3 bucket,
+ * and generates an array of filtered objects for each row of data.
+ *
+ * @async
+ * @function parseCsvToArray
+ * @param {string} csvString - The CSV data as a string.
+ * @returns {Array} An array of objects that passed the filter function.
+ * @throws {Error} If an error occurs during the CSV processing.
```
Comment (chenglawson1) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:188]: No error is thrown in this function.

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 9]
```diff
@@ -163,7 +206,14 @@ export const parseCsvToArray = async (csvString) => {
   });
 };
 
-// returns unique records array and an array that contains all the duplicate
+/**
+ * Filters unique entries based on NHS number.
+ *
+ * @function filterUniqueEntries
+ * @param {Array<Object>} caasFeed - The array of records to filter.
+ * @returns {Array<Object>} An array of unique records and an array that contains all the duplicate.
```
Comment (chenglawson1) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:214]: I think the return type is an Array of 2 arrays.

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 10]
```diff
@@ -180,7 +230,18 @@ export const filterUniqueEntries = (cassFeed) => {
...
+ * Check if the nhs_number of the record is present in Population Table.
+ * If yes, is is rejected. If no, its superseded_by_nhs_number value is checked.
+ * If thats null, record is added to table, else
+ * Check if the superseded_by_nhs_number of the record is present in Population Table.
+ * If yes, is is rejected. else added to table.
+ *
+ * @function processingData
+ * @param {Object} record - details of participant.
+ * @returns {Object} returns formatted dynamodb record or rejected record object.
+ */
```
Comment (chenglawson1) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:243]: Please add @async.

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 11]
```diff
@@ -218,7 +279,19 @@ const processingData = async (record) => {
...
+ *
+ * @function checkDynamoTable
+ * @param {DynamoDBClient} dbClient Instance of DynamoDB client
+ * @param {string} attribute Value of the field in the DynamoDB table
+ * @param {string} table Name of the DynamoDB table
+ * @param {string} attributeName Name of the field in the DynamoDB table
+ * @param {string} attributeType Type of the field in the DynamoDB table
+ * @param {boolean} useIndex tells if the DynamoDB search need to be made on index or key.
+ * @returns {boolean} True if the attribute exists in the DynamoDB table, otherwise false.
+ */
```
Comment (chenglawson1) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:293]: Please add @async.

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 12]
```diff
@@ -280,8 +360,16 @@ const generateRecord = async (record, client) => {
...
-// returns unique participant id using NHS regex
-// unique in both population table and this stream
+/**
+ * Generates unique participant id using NHS regex
+ * unique in both population table and this stream
+ * @async
+ * @function generateParticipantID
+ * @param {DynamoDBClient} dbClient Instance of DynamoDB client
+ * @returns {String} participant id
+ * @throws {Error} Error: generating participant id.
```
Comment (chenglawson1) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:370]: Error is not thrown, it is returned. So return type should be changed to string|Error

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 13]
```diff
@@ -399,7 +506,18 @@ export const getItemFromTable = async (dbClient, table, ...keys) => {
...
+/**
+ * This function allows the user to query against DynamoDB.
+ * @async
+ * @function lookUp
+ * @param {DynamoDBClient} dbClient Instance of DynamoDB client
+ * @param  {...any} params params is destructed to id, which is the value you use to query against.
+ * The table is the table name (type String), attribute is the column you search against (type String),
+ * attributeType is the type of data stored within that column and useIndex is toggled to true if you want to use
+ * an existing index (type boolean)
+ * @returns {boolean} returns successful response if attribute doesn't exist in table
```
Comment (chenglawson1) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:518]: Return type is a number, 200 or 400.

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 14]
```diff
@@ -449,7 +579,17 @@ export const uploadToDynamoDb = async (dbClient, table, batch) => {
...
-// returns array with each element representing response of batch upload status
+/**
+ * Writes the records to DynamoDB
+ *
+ * @async
+ * @function batchWriteRecords
+ * @param {Object} record - records to be written to DynamoDB.
+ * @param {Number} chunkSize size of the chunks.
+ * @param {DynamoDBClient} dbClient Instance of DynamoDB client
+ * @returns {Array<Object>} returns array with each element representing response of batch upload status.
```
Comment (chenglawson1) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:590]: Array of Promise.

**Comment by NeerjaGoyalNHS** [comment]
Added to PR: https://github.com/NHSDigital/Galleri-Invitations/pull/663/files

**Pull request closed without merging**